Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -49,9 +50,25 @@ public override void Initialize()
             SubscribeLocalEvent<GuardianHostComponent, MobStateChangedEvent>(OnHostStateChange);
             SubscribeLocalEvent<GuardianHostComponent, ComponentShutdown>(OnHostShutdown);
 
+            SubscribeLocalEvent<GuardianHostComponent, PerformInstantActionEvent>(OnPerformAction);
+
             SubscribeLocalEvent<GuardianComponent, AttackAttemptEvent>(OnGuardianAttackAttempt);
         }
 
+        private void OnPerformAction(EntityUid uid, GuardianHostComponent component, PerformInstantActionEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            if (args.Action != component.Action)
+                return;
```

Comment:
still a draft ik but im not sure about the general idea of how these work now, having stuff subscribe to every action event like this and manually checking if its the right action is a little weird. plus defining the actions entirely in components seems weird, i feel like it should just point to a yaml prototype id that actually contains the data or something? i'm not sure. this is definitely a lot better than before though

Reply:
Yeah I agree the whole thing with subbing to all action executions and then checking the action was weird. I switched to making actions just raise yaml-defined events. IMO that set up feels nicer.

I'm still somewhat iffy on how components need to keep track of their actions in order to keep the component-state and action-state in sync (i.e., is flashlight toggled on).

In terms of making more component actions use prototypes to define their actions, but for the sake of tracking togggling & cooldowns per-component, each one component would still need to create an instance of the action protoype on init

Reply:
II can't currently think of a nice way to do toggleable actions defined only with prototype strings. But maybe I'll think of something later.